site/app/libraries/database/DatabaseQueries.php

```diff
@@ -2518,25 +2523,42 @@ public function getGradeableConfig($g_id) {
                 eg_late_days AS late_days,
                 eg_allow_late_submission AS late_submission_allowed,
                 eg_precision AS precision
-            FROM electronic_gradeable WHERE g_id=?";
+              FROM electronic_gradeable
+            ) AS eg ON g.g_id=eg.eg_id
+            WHERE g.g_id=?";
         $this->course_db->query($query, array($g_id));
 
-        $elec_details = $this->course_db->row();
-        if(count($elec_details) === 0 && $details['type'] === GradeableType::ELECTRONIC_FILE) {
-            throw new DatabaseException("Electronic gradeable didn't have an entry in the electronic_gradeable table!");
+        $details = $this->course_db->row();
+        if(count($details) === 0) {
+            throw new \InvalidArgumentException("Gradeable ID did not exist!");
         }
 
-        $details = array_merge($details, $elec_details);
+        if(!isset($details['eg_id']) && $details['type'] === GradeableType::ELECTRONIC_FILE) {
+            throw new DatabaseException("Electronic gradeable didn't have an entry in the electronic_gradeable table!");
+        }
 
         // Finally, create the gradeable
         $gradeable = new \app\models\gradeable\Gradeable($this->core, $details, []);
 
         // Get the components data
         $gradeable->setComponents($this->getGradeableComponentConfigs($gradeable));
 
+        // Set the rotating sections data
+        $gradeable->setRotatingGraderSections($this->getGradersForAllRotatingSections($g_id));
+
         return $gradeable;
```

Comment:
`setRotatingGraderSections` marks the `Gradeable` as modified before it even returns from this function. You should `setModified(false)` it at the end or move those setting functions into its constructor before you unset `modified`.